### PR TITLE
A utility you can use to combine an `AbortSignal` and a `Promise` into an abortable promise

### PR DESCRIPTION
**Pull request by steveluscher**

### CODE REVIEW

**Comment by steveluscher** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2791?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#2772** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2772?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2791** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2791?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2790** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2790?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @steveluscher and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/react/src/abortable-promise.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+export function getAbortablePromise<T>(promise: Promise<T>, abortSignal?: AbortSignal): Promise<T> {
+    if (!abortSignal) {
+        return promise;
```
[the remark this answers is not in the document]
Comment (steveluscher) [reply packages/react/src/abortable-promise.ts:3]: I know. It just depends where you want to move the complexity. You either get this, or you get it at each caller:

```ts
// This...
const stuffPromise = doStuff();
const stuff = await (abortSignal
  ? getAbortablePromise(stuffPromise, abortSignal)
  : stuffPromise);

// Versus...
const stuff = await getAbortablePromise(doStuff(), abortSignal);
```

[the remark this answers is not in the document]
Comment (buffalojoec) [reply packages/react/src/abortable-promise.ts:3]: Yeah, I figured this was why you did it. Is this internal-only throughout the rest of your stack?

[the remark this answers is not in the document]
Comment (steveluscher) [reply packages/react/src/abortable-promise.ts:3]: It is, yeah.

[the remark this answers is not in the document]
Comment (buffalojoec) [reply packages/react/src/abortable-promise.ts:3]: Cool, then wfm!

**Comment by steveluscher** [comment]
### Merge activity

* **Jun 11, 12:22 AM PDT**: @steveluscher started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2791).

**Pull request merged**